TestApp/Booking/Sources/BookingViews/BookingScreen.swift

```diff
@@ -0,0 +1,58 @@
+import SwiftUI
+import Navigation
+
+public struct BookingScreen: View {
+
+    enum Modal {
+        case dummyAlert
+        case dummySheet
+    }
+
+    @Environment(\.presentationMode) var presentationMode
+    @StateObject var bookingDismissalConfirmation = BookingDismissalConfirmationModel()
+    @State var modal: Modal?
+
+    public let token: String
+    public let finish: (_ bookingID: String) -> Void
+
+    public var body: some View {
+        NavigationView {
+            VStack(spacing: 20) {
+                Text("Token: \(token)")
+
+                Button("Show dummy alert") {
+                    modal = .dummyAlert
+                }
+
+                Button("Show dummy sheet") {
+                    modal = .dummySheet
+                }
+
+                Button("Go to booking detail in MMB") {
+                    finish("123")
+                }
+
+                Button("Dismiss") {
+                    Task {
+                        if await bookingDismissalConfirmation.canBeDismissed() {
+                            presentationMode.wrappedValue.dismiss()
```

Comment:
As the `presentationMode` is obsolete, here we can use `dismiss()` environment value directly

Reply:
Sure, I think this originally supported iOS 13...

Reply:
> 'dismiss' is only available in iOS 15.0 or newer